### PR TITLE
Improve Signaling Logic

### PR DESCRIPTION
**Pull request by pojntfx**
This improves the channels used to signal ready states by using `chan struct{}` instead of `chan any` consistently. While `chan any` allows for sending arbitrary data, `chan struct{}` allows sending no data, which makes more sense if the only purpose is to `close` the channel to signal anyways.

### CODE REVIEW

**Review by ShivanshVij — APPROVED** [review]
LGTM! `chan struct{}` is also great because it doesn't allocate any memory.

File: os/configs/drafteros-firecracker-x86_64_pvm_defconfig [hunk 1]
```diff
@@ -1,6 +1,7 @@
 BR2_x86_64=y
 BR2_KERNEL_HEADERS_VERSION=y
-BR2_DEFAULT_KERNEL_VERSION="6.8.12"
+BR2_DEFAULT_KERNEL_VERSION="6.7.0-rc6"
```
Comment (ShivanshVij) [inline os/configs/drafteros-firecracker-x86_64_pvm_defconfig:3]: Wait why is this change here?

Comment (pojntfx) [reply os/configs/drafteros-firecracker-x86_64_pvm_defconfig:3]: Bug fix! Didn't notice it earlier since CI/CD was down, but the newer version of Buildroot actually checks the kernel header versions, so I've specified the proper kernel release string :)
